src/edu/psu/sweng500/team8/coreDataStructures/Cell.java

```diff
@@ -104,4 +104,41 @@ public void load(BinaryInputStream stream) throws IOException {
 			this.type = ValueType.Given;
 		}
 	}
+
+	@Override
+	public boolean equals(Object o) {

```

Comment:
I see you're using this to check for duplicate SudokuActions in the undo stack, but this seems like this is just covering up that bug. Can't we fix that instead?